client/virtualfund_integration_test.go

```diff
@@ -0,0 +1,121 @@
+
+	bKey := common.Hex2Bytes(`0279651921cd800ac560c21ceea27aab0107b67daf436cdd25ce84cad30159b4`)
+	b := common.HexToAddress(`0xBBB676f9cFF8D242e9eaC39D063848807d3D1D94`)
+
+	iKey := common.Hex2Bytes(`febb3b74b0b52d0976f6571d555f4ac8b91c308dfa25c7b58d1e6a7c3f50c781`)
+	i := common.HexToAddress(`0x111A00868581f73AB42FEEF67D235Ca09ca1E8db`)
+
+	chain := chainservice.NewMockChain([]types.Address{a, b, i})
+
+	chainservA := chainservice.NewSimpleChainService(chain, a)
+	messageserviceA := messageservice.NewTestMessageService(a)
+	storeA := store.NewMockStore(aKey)
+	clientA := New(messageserviceA, chainservA, storeA, logDestination)
+
+	chainservB := chainservice.NewSimpleChainService(chain, b)
+	messageserviceB := messageservice.NewTestMessageService(b)
+	storeB := store.NewMockStore(bKey)
+	clientB := New(messageserviceB, chainservB, storeB, logDestination)
+
+	chainservI := chainservice.NewSimpleChainService(chain, i)
+	messageserviceI := messageservice.NewTestMessageService(i)
+	storeI := store.NewMockStore(iKey)
+	clientI := New(messageserviceI, chainservI, storeI, logDestination)
+
+	messageserviceA.Connect(messageserviceB)
+	messageserviceA.Connect(messageserviceI)
+
+	messageserviceB.Connect(messageserviceA)
+	messageserviceB.Connect(messageserviceI)
+
+	messageserviceI.Connect(messageserviceA)
+	messageserviceI.Connect(messageserviceB)
+
+	directlyFundALedgerChannel := func(alpha Client, beta Client) {
+		// Set up an outcome that requires both participants to deposit
+		outcome := outcome.Exit{outcome.SingleAssetExit{
+			Allocations: outcome.Allocations{
+				outcome.Allocation{
+					Destination: types.AddressToDestination(*alpha.Address),
+					Amount:      big.NewInt(5),
+				},
+				outcome.Allocation{
+					Destination: types.AddressToDestination(*beta.Address),
+					Amount:      big.NewInt(5),
+				},
+			},
+		}}
+		id := alpha.CreateDirectChannel(*beta.Address, types.Address{}, types.Bytes{}, outcome, big.NewInt(0))
+		waitForCompletedObjectiveId(id, &alpha)
+		waitForCompletedObjectiveId(id, &beta)
+
+	}
+
+	directlyFundALedgerChannel(clientA, clientI)
+	directlyFundALedgerChannel(clientI, clientB)
+
+	outcome := outcome.Exit{outcome.SingleAssetExit{
+		Allocations: outcome.Allocations{
+			outcome.Allocation{
+				Destination: types.AddressToDestination(a),
+				Amount:      big.NewInt(5),
+			},
+			outcome.Allocation{
+				Destination: types.AddressToDestination(b),
+				Amount:      big.NewInt(5),
+			},
+		},
+	}}
+	id := clientA.CreateVirtualChannel(b, i, types.Address{}, types.Bytes{}, outcome, big.NewInt(0))
+	waitForCompletedObjectiveId(id, &clientA)
+	waitForCompletedObjectiveId(id, &clientB)
+	waitForCompletedObjectiveId(id, &clientI)
+
+}
+
+// waitForCompletedObjectiveId waits for completed objectives and returns when the completed objective id matchs the id waitForCompletedObjectiveId has been given
+func waitForCompletedObjectiveId(id protocols.ObjectiveId, client *Client) {
```

Comment:
This integration test asserts that the objective is completed, but does no assertions on the end result. Is this cause for concern?

Reply:
Unfortunately the current API doesn't expose anything beyond the completed `ObjectiveId`. For no we could probably interrogate the `store` on the client to make sure states look like we expect them to.